dispatcher/backend/maint-scripts/create_ted_topics_recipes.py

```diff
@@ -0,0 +1,152 @@
+        SEARCH_URL,
+        headers={"User-Agent": "Mozilla/5.0"},
+        json=data,
+        timeout=REQUESTS_TIMEOUT,
+    )
+    req.raise_for_status()
+    topics = json.loads(req.content)["results"][0]["facets"]["tags"].keys()
+    logger.info(f"{len(topics)} topics found")
+    return topics
+
+
+def create_recipe(ted_topic_name: str, access_token):
+    clean_ted_topic_name = ted_topic_name.replace(" ", "-")
+    schedule_name = f"ted_topic_{clean_ted_topic_name}"
+    response = requests.get(
+        get_url(f"/schedules/{schedule_name}"),
+        headers=get_token_headers(access_token),
+    )
+    if response.status_code == HTTPStatus.OK:
+        logger.warning(f"Recipe {schedule_name} already exists, ignoring.")
+        return
+
+    if response.status_code != HTTPStatus.NOT_FOUND:
+        response.raise_for_status()
+
+    logger.info(f"Creating recipe for {ted_topic_name}")
+
+    data = {
+        "category": "ted",
+        "config": {
+            "flags": {
+                "description": f"A collection of TED videos about {ted_topic_name}",
+                "title": f"TED {ted_topic_name}",
+                "topics": ted_topic_name,
+                "name": f"ted_mul_{clean_ted_topic_name}",
+                "languages": "eng,fra,esp,deu,chi",
+                "format": "webm",
+                "low-quality": True,
+                "optimization-cache": os.environ["TED_OPTIM_CACHE_URL"],
+                "output": "/output",
+                "tmp-dir": "/output",
+                "subtitles": "all",
+                "subtitles-enough": True,
+                "debug": True,
+                "tags": "ted",
+            },
+            "image": {
+                "name": "ghcr.io/openzim/ted",
+                "tag": "2.1.0",
+            },
+            "monitor": False,
+            "platform": "ted",
+            "resources": {
+                "cpu": 3,
+                "disk": 32212254720,
+                "memory": 2147483648,
+            },
+            "task_name": "ted",
+            "warehouse_path": "/.hidden/dev",
+        },
+        "enabled": True,
+        "language": {
+            "code": "mul",
+            "name_en": "Multiple Languages",
+            "name_native": "Multiple Languages",
+        },
+        "name": schedule_name,
+        "periodicity": "monthly",
+        "tags": [
+            "ted-by-topic",
+        ],
+    }
+    response = requests.post(
+        get_url("/schedules/"),
+        headers=get_token_headers(access_token),
+        json=data,
+    )
+    if response.status_code != HTTPStatus.CREATED:
+        logger.error(json.loads(response.content))
+        response.raise_for_status()
+
+
+def main(zf_username, zf_password):
+    """Creates recipes for TED by topics"""
+
+    access_token, refresh_token = get_token(zf_username, zf_password)
+
+    ted_topics = get_ted_topics()
+    logger.debug(ted_topics)
+    for topic in ted_topics:
+        if topic != "street art":
```

Comment:
This is the only one which is going to be executed in fact.

I decided to keep this in the script as a foolproof thing to ensure the next person who will use the script will understand how it works before launching it, and if not it will impact only one topic/recipe